src/plugins/vis_builder/public/application/components/data_tab/field_selector_field.tsx

```diff
@@ -28,59 +28,119 @@
  * under the License.
  */
 
-import React, { useState } from 'react';
-import { IndexPatternField } from '../../../../../data/public';
-import { FieldButton, FieldIcon } from '../../../../../opensearch_dashboards_react/public';
-import { useDrag } from '../../utils/drag_drop/drag_drop_context';
-import { COUNT_FIELD } from '../../utils/drag_drop/types';
+import React, { useCallback, useState } from 'react';
+import { EuiPopover } from '@elastic/eui';
 
+import {
+  FilterManager,
+  IndexPattern,
+  IndexPatternField,
+  opensearchFilters,
+} from '../../../../../data/public';
+import {
+  FieldButton,
+  FieldButtonProps,
+  FieldIcon,
+} from '../../../../../opensearch_dashboards_react/public';
+
+import { COUNT_FIELD, useDrag } from '../../utils/drag_drop';
+import { VisBuilderFieldDetails } from './field_details';
+import { FieldDetails } from './types';
 import './field_selector_field.scss';
 
 export interface FieldSelectorFieldProps {
-  field: Partial<IndexPatternField> & Pick<IndexPatternField, 'displayName' | 'type' | 'scripted'>;
+  field: IndexPatternField;
+  filterManager: FilterManager;
+  indexPattern?: IndexPattern;
+  getDetails: (field) => FieldDetails;
 }
 
-// TODO:
-// 1. Add field sections (Available fields, popular fields from src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx)
-// 2. Add popover for fields stats from discover as well
-export const FieldSelectorField = ({ field }: FieldSelectorFieldProps) => {
+// TODO: Add field sections (Available fields, popular fields from src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx)
+export const FieldSelectorField = ({
+  field,
+  filterManager,
+  indexPattern,
+  getDetails,
+}: FieldSelectorFieldProps) => {
+  const { id: indexPatternId = '', metaFields = [] } = indexPattern ?? {};
+  const isMetaField = metaFields.includes(field.name);
   const [infoIsOpen, setOpen] = useState(false);
-  const [dragProps] = useDrag({
-    namespace: 'field-data',
-    value: field.name || COUNT_FIELD,
-  });
+
+  const onAddFilter = useCallback(
+    (fieldToFilter, value, operation) => {
+      const newFilters = opensearchFilters.generateFilters(
+        filterManager,
+        fieldToFilter,
+        value,
+        operation,
+        indexPatternId
+      );
+      return filterManager.addFilters(newFilters);
+    },
+    [filterManager, indexPatternId]
+  );
 
   function togglePopover() {
     setOpen(!infoIsOpen);
   }
 
+  return (
+    <EuiPopover
+      ownFocus
+      display="block"
+      button={<SelectorFieldButton isActive={infoIsOpen} onClick={togglePopover} field={field} />}
+      isOpen={infoIsOpen}
+      closePopover={() => setOpen(false)}
+      anchorPosition="rightUp"
+      panelClassName="vbItem__fieldPopoverPanel"
+      repositionOnScroll
```

Comment:
@ashwin-pc This doesn't appear to work as I would expect - I could use a hand in fixing the behavior when the field list scrolls.

Reply:
I was just going to comment on this. So syncing on scroll is expected, nice. Do we need this for 2.4 in your opinion? Like it is not pleasant, but I dont want to block this feature over it

Reply:
Yeah, seems like something that's intended to work out of the box from EUI/OUI. In my opinion, the new functionalities we have here are worth getting into 2.4 even with this behavior. But yeah, it's obnoxious - at least worth some timeboxed investigation.

Reply:
This may be an issue for OUI. I think this relies on whole page scroll and may not be something that we can fix on our end

Reply:
Yep: https://github.com/opensearch-project/oui/blob/main/src/components/popover/popover.tsx#L535